From b1531a6b830c7ab31c84098c1b0bf937d60eea29 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 18 Nov 2021 15:16:31 +0000
Subject: [PATCH 1/1] Disable clone3 wrapper

Firefox's and Chromium's sandboxes had to be patched to support this properly. Getting
the chromium patch rolled out to all the various electron apps will take a loong
time. Otherwise we just segfault.
---
 sysdeps/unix/sysv/linux/clone-internal.c | 2 +-
 sysdeps/unix/sysv/linux/i386/sysdep.h    | 2 +-
 sysdeps/unix/sysv/linux/x86_64/sysdep.h  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/clone-internal.c b/sysdeps/unix/sysv/linux/clone-internal.c
index 979f7880..443564c7 100644
--- a/sysdeps/unix/sysv/linux/clone-internal.c
+++ b/sysdeps/unix/sysv/linux/clone-internal.c
@@ -48,7 +48,7 @@ __clone_internal (struct clone_args *cl_args,
 		  int (*func) (void *arg), void *arg)
 {
   int ret;
-#ifdef HAVE_CLONE3_WRAPPER
+#if 0
   /* Try clone3 first.  */
   int saved_errno = errno;
   ret = __clone3 (cl_args, sizeof (*cl_args), func, arg);
diff --git a/sysdeps/unix/sysv/linux/i386/sysdep.h b/sysdeps/unix/sysv/linux/i386/sysdep.h
index ec12c84c..0d4d7474 100644
--- a/sysdeps/unix/sysv/linux/i386/sysdep.h
+++ b/sysdeps/unix/sysv/linux/i386/sysdep.h
@@ -290,7 +290,7 @@ struct libc_do_syscall_args
 # define HAVE_TIME_VSYSCALL             "__vdso_time"
 # define HAVE_CLOCK_GETRES_VSYSCALL     "__vdso_clock_getres"
 
-# define HAVE_CLONE3_WRAPPER		1
+# define HAVE_CLONE3_WRAPPER		0
 
 # undef HAVE_INTERNAL_BRK_ADDR_SYMBOL
 # define HAVE_INTERNAL_BRK_ADDR_SYMBOL 1
diff --git a/sysdeps/unix/sysv/linux/x86_64/sysdep.h b/sysdeps/unix/sysv/linux/x86_64/sysdep.h
index 327e5938..572f4a04 100644
--- a/sysdeps/unix/sysv/linux/x86_64/sysdep.h
+++ b/sysdeps/unix/sysv/linux/x86_64/sysdep.h
@@ -377,7 +377,7 @@
 # define HAVE_GETCPU_VSYSCALL		"__vdso_getcpu"
 # define HAVE_CLOCK_GETRES64_VSYSCALL   "__vdso_clock_getres"
 
-# define HAVE_CLONE3_WRAPPER			1
+# define HAVE_CLONE3_WRAPPER			0
 
 # define SINGLE_THREAD_BY_GLOBAL		1
 
-- 
2.34.0

